starter-code/api/routes/users.js

```diff
@@ -1,9 +1,71 @@
 var express = require('express');
 var router = express.Router();
+//var ObjectId = require('mongoose').Types.ObjectId;
+const { body, validationResult } = require('express-validator');
+
+const Users = require('../models/users');
+
+router.get("/", function(req, res, next) {
+  const body = req.body;
+  Users.find({}, (err, users) => {
+      return res.status(200).json(users).end();
+  })
 
-/* GET users listing. */
-router.get('/', function(req, res, next) {
-  res.send('respond with a resource');
 });
 
+router.post("/",  
+    body("full_name")
+        .exists().withMessage('full_name is required.').bail()
+        .notEmpty().withMessage('full_name is required.').bail(), 
+    body("given_name")
+        .exists().withMessage('given_name is required.').bail()
+        .notEmpty().withMessage('given_name is required.').bail(), 
+    body("family_name")
+        .exists().withMessage('family_name is required.').bail()
+        .notEmpty().withMessage('family_name is required.').bail(), 
+    body("email")
+        .exists().withMessage('email is required.').bail()
+        .notEmpty().withMessage('email is required.').bail(), 
+
+    function(req, res, next) {
+        const errors = validationResult(req);
+        const body = req.body;
+
+        /*Users.findOne({email: body.email}, (err, user) => {
+            exists = true;
+        });*/
+
+       Users.find({email : body.email}, function (err, docs) {
+            let exists = false;
+            if (docs.length){
+                exists = true;
+            }
+
+            if (!errors.isEmpty() || exists) {
+                return res.status(400).end();
```

Comment:
```suggestion
                return res.status(400).json({errors: errors.array()});
```
So the client can see error messages on faulty inputs.